src/simplewallet/simplewallet.cpp

```diff
@@ -3489,6 +3490,86 @@ bool simple_wallet::print_integrated_address(const std::vector<std::string> &arg
   return true;
 }
 //----------------------------------------------------------------------------------------------------
+bool simple_wallet::address_book(const std::vector<std::string> &args/* = std::vector<std::string>()*/)
+{
+  if (args.size() == 0)
+  {
+  }
+  else if (args.size() == 1 || (args[0] != "add" && args[0] != "delete"))
+  {
+    fail_msg_writer() << tr("usage: address_book [(add (<address> [pid <long or short payment id>])|<integrated address> [<description possibly with whitespaces>])|(delete <index>)]");
+    return true;
+  }
+  else if (args[0] == "add")
+  {
+    cryptonote::account_public_address address;
```

Comment:
Not sure if that's going to be very useful. You can simply query any OpenAlias address by `transfer` (then cancel it), and add an entry to the address book with the alias as its description. Otherwise, the GUI should accommodate the new feature first, in my opinion.